packages/expect/src/utils.ts

```diff
@@ -95,13 +102,18 @@ export function wrapAssertion(
     }
 
     try {
-      return fn.apply(this, args)
+      const result = fn.apply(this, args)
+
+      if (result && typeof result === 'object' && result instanceof Promise) {
```

Comment:
I thought I already fixed that, but maybe I forgot. Can you change the check from `instanceof Promise` to `typeof result.then === 'function'` here and in `jest-extend.ts`? 

And use `then(noop, (err) => { /* ... */ })` here instead